runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7591,10 +7595,18 @@ done:;
 			}
 			goto retry;
 		}
-		/* Unresolved */
+		/* Unresolved or requires initialization in the VT case */
 		buildGenericSpecialStackFrame(REGISTER_ARGS, 0);
 		updateVMStruct(REGISTER_ARGS);
-		resolveClassRef(_currentThread, ramConstantPool, index, J9_RESOLVE_FLAG_RUNTIME_RESOLVE);
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		if (NULL == resolvedClass) {
+#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
+			resolveClassRef(_currentThread, ramConstantPool, index, J9_RESOLVE_FLAG_RUNTIME_RESOLVE);
```

Comment:
You probably want to pass in `J9_RESOLVE_FLAG_INIT_CLASS` as well here to avoid going around the loop again.